pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -457,13 +460,47 @@ var (
 	errNewServiceAccount = errors.New("service account change")
 )
 
+// WorkloadUpdate is called when workload labels/annotations are updated.
+func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, annotations map[string]string) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	w, f := s.WorkloadsById[id]
+	if !f {
+		// First time this workload has been seen. Likely never connected, no need to
```

Comment:
? why not? Others need to know that this new workload appeared

Reply:
I didn't find any use - once the endpoint shows up in a service there is a notification. 
Also when the workload connects - this is looked up.

I don't mind a new notification - but not sure what is the case, we don't have it in current code ( workload showing up doesn't result in cache invalidation)

Reply:
? doesn't it trigger EDS updates (incremental) for all clusters where this workload is part of?